File: src/algorithms/strings/transformation/run-length-decoding/sources/RunLengthDecoding_test.cpp
/** Correctness tests for the runLengthDecoding function. */
#include "RunLengthDecoding.cpp"
#include <cassert>
#include <iostream>

int main() {
    assert(runLengthDecoding("3a2b4c") == "aaabbcccc");
    assert(runLengthDecoding("1a1b1c") == "abc");
    assert(runLengthDecoding("") == "");
    assert(runLengthDecoding("1z") == "z");
    assert(runLengthDecoding("5x") == "xxxxx");
    assert(runLengthDecoding("2a3b1c") == "aabbbc");
    assert(runLengthDecoding("10a") == "aaaaaaaaaa");
    assert(runLengthDecoding("2a2a") == "aaaa");
    assert(runLengthDecoding("3A2B") == "AAABB");
    std::cout << "All tests passed!" << std::endl;
    return 0;
}

File: src/algorithms/strings/transformation/run-length-decoding/__tests__/RunLengthDecoding_test.java
/** Correctness tests for the RunLengthDecoding algorithm. */
public class RunLengthDecoding_test {
    public static void main(String[] args) {
        assert RunLengthDecoding.runLengthDecoding("3a2b4c").equals("aaabbcccc");
        assert RunLengthDecoding.runLengthDecoding("1a1b1c").equals("abc");
        assert RunLengthDecoding.runLengthDecoding("").equals("");
        assert RunLengthDecoding.runLengthDecoding("1z").equals("z");
        assert RunLengthDecoding.runLengthDecoding("5x").equals("xxxxx");
        assert RunLengthDecoding.runLengthDecoding("2a3b1c").equals("aabbbc");
        assert RunLengthDecoding.runLengthDecoding("10a").equals("aaaaaaaaaa");
        assert RunLengthDecoding.runLengthDecoding("2a2a").equals("aaaa");
        assert RunLengthDecoding.runLengthDecoding("3A2B").equals("AAABB");
        System.out.println("All tests passed!");
    }
}
